src/components/structures/RoomView.js

```diff
@@ -205,6 +205,8 @@ module.exports = React.createClass({
 
         this.setState({
             events: [],
+            searchResults: null, // we may have arrived here by clicking on a
+                                 // search result. Hide the results.
```

Comment:
I don't like how loading the timeline randomly nulls out search results, but I think this is more a problem with `RoomView` than anything else.